server/icat/include/irods/icatHighLevelRoutines.hpp

```diff
@@ -363,4 +363,21 @@ auto chl_get_delay_rule_info(RsComm& _comm, const char* _rule_id, std::vector<st
 /// \since 4.2.12
 auto chl_data_object_finalize(RsComm& _comm, const char* _json_input) -> int;
 
+/// \brief High-level wrapper for fetching all the information about a delay rule.
+///
+/// Triggers policy associated with database operations.
+///
+/// \param[in] _comm     The communication object.
+/// \param[in] _username The part of a fully-qualified iRODS username preceding the pound sign.
+/// \param[in] _zone     The part of a fully-qualified iRODS username following the pound sign.
+/// \param[in] _password The expected obfuscated password.
+///
+/// \return An integer.
+/// \retval 1                 If the obfuscated password matches what is stored in the catalog.
+/// \retval CAT_NO_ROWS_FOUND If no match was found in the catalog.
+/// \retval <0                If an error occurred.
+///
+/// \since 4.3.1
+auto chl_check_auth_credentials(RsComm& _comm, const char* _username, const char* _zone, const char* _password) -> int;
```

Comment:
Consider taking an out parameter instead of using the return type.

While this works, the documentation here can be misleading because `CAT_NO_ROWS_FOUND` is a negative integer.

Reply:
Updated to take an additional parameter.